src/software/ai/hl/stp/skill/dribble/dribble_skill_fsm.h

```diff
@@ -0,0 +1,220 @@
+     *
+     * @param event the Update event
+     */
+    void dribble(const Update &event);
+
+    /**
+     * Action to lose control of the ball
+     *
+     * @param event the Update event
+     */
+    void loseBall(const Update &event);
+
+    /**
+     * Guard that checks if the robot has control of the ball
+     *
+     * @param event the Update event
+     *
+     * @return if the robot has control of the ball
+     */
+    bool haveBallControl(const Update &event);
+
+    /**
+     * Guard that checks if the robot has lost control of the ball
+     *
+     * @param event the Update event
+     *
+     * @return if the robot has lost control of the ball
+     */
+    bool lostBallControl(const Update &event);
+
+    /**
+     * Guard that checks if the ball is at the dribble_destination and robot is facing
+     * the right direction with control of the ball
+     *
+     * @param event the Update event
+     *
+     * @return if the ball is at the dribble_destination, robot is facing the correct
+     * direction and has control of the ball
+     */
+    bool dribblingDone(const Update &event);
+
+    /**
+     * Guard that checks if the the robot should lose control of the ball to avoid
+     * excessive dribbling
+     *
+     * @param event the Update event
+     *
+     * @return if the ball should be lost
+     */
+    bool shouldLoseBall(const Update &event);
+
+    /**
+     * Guard that checks whether the excessive_dribbling_mode control param
+     * is NOT set to TbotsProto::ExcessiveDribblingMode::TERMINATE
+     *
+     * @param event the Update event
+     *
+     * @return false if the excessive_dribbling_mode control param is set
+     * to TbotsProto::ExcessiveDribblingMode::TERMINATE, true otherwise
+     */
+    bool shouldExcessivelyDribble(const Update &event);
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        DEFINE_SML_STATE(GetBallControl)
+        DEFINE_SML_STATE(Dribble)
+        DEFINE_SML_STATE(LoseBall)
+
+        DEFINE_SML_EVENT(Update)
+
+        DEFINE_SML_GUARD(haveBallControl)
+        DEFINE_SML_GUARD(lostBallControl)
+        DEFINE_SML_GUARD(dribblingDone)
+        DEFINE_SML_GUARD(shouldLoseBall)
+        DEFINE_SML_GUARD(shouldExcessivelyDribble)
+
+        DEFINE_SML_ACTION(loseBall)
+        DEFINE_SML_ACTION(getBallControl)
+        DEFINE_SML_ACTION(dribble)
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *GetBallControl_S + Update_E[haveBallControl_G] / dribble_A = Dribble_S,
+            GetBallControl_S + Update_E / getBallControl_A,
+
+            Dribble_S + Update_E[lostBallControl_G] / getBallControl_A = GetBallControl_S,
+            Dribble_S + Update_E[shouldLoseBall_G && shouldExcessivelyDribble_G] /
+                            loseBall_A = LoseBall_S,
+            Dribble_S +
+                Update_E[shouldLoseBall_G && !shouldExcessivelyDribble_G] / dribble_A = X,
```

Comment:
```suggestion
            Dribble_S +
                Update_E[shouldLoseBall_G && !shouldExcessivelyDribble_G] / loseBall_A = X,
```
If we don't want to excessively dribble, should we have `loseBall` here?